File: leetcode.com/1470_shuffle_the_array/solution.go
func shuffle(nums []int, n int) []int {
    a := make([]int, n*2)

    for i := 0 ; i < n ; i++ {
        a[i*2] = nums[i]
        a[i*2+1] = nums[n + i]
    }

    return a
}

File: leetcode.com/1470_shuffle_the_array/solution.c
int *shuffle(int *nums, int numsSize, int n, int *returnSize)
{
    int *a = calloc(sizeof(int), numsSize);

    for (int i = 0 ; i < n ; i++) {
        a[i*2] = nums[i];
        a[i*2+1] = nums[n+i];
    }

    *returnSize = numsSize;
    return a;
}
